src/helpers/getLocaleFromRequest/index.js

```diff
@@ -11,12 +11,20 @@ function getAcceptedLanguage (request) {
   return acceptedLanguages[0]
 }
 
+function getQueryFromRequest (request) {
+  return function (param, key) {
+    return decodeURIComponent(get(request, `query.${key}`)) || param
+  }
+}
+
 function getLocaleFromRequest (options) {
   var cookieName = get(options, 'cookieName', 'locale')
+  var params = get(options, 'queryParams', ['locale'])
+  var queryParams = typeof params === 'string' ? [params] : params
 
   return function (request) {
     var locale = (
-      decodeURIComponent(get(request, 'query.locale', '')) ||
+      queryParams.reduce(getQueryFromRequest(request), null) ||
```

Comment:
FYI @hilja – we did need the default value here, because reduce as a default takes the first argument of the array as the default, so the key was always being returned